src/modules/Elsa.Workflows.Runtime/Services/BackgroundActivityInvoker.cs

```diff
@@ -90,35 +90,8 @@ private async Task ResumeWorkflowAsync(ActivityExecutionContext activityExecutio
     }
 
     private IDictionary<string, object> ExtractActivityOutput(ActivityExecutionContext activityExecutionContext)
-    {
-        var outputDescriptors = activityExecutionContext.ActivityDescriptor.Outputs;
-        var outputValues = new Dictionary<string, object>();
-
-        foreach (var outputDescriptor in outputDescriptors)
-        {
-            var output = (Output?)outputDescriptor.ValueGetter(activityExecutionContext.Activity);
-
-            if (output == null)
-                continue;
-
-            var memoryBlockReference = output.MemoryBlockReference();
-
-            if (!activityExecutionContext.ExpressionExecutionContext.TryGetBlock(memoryBlockReference, out var memoryBlock))
-                continue;
-
-            var variableMetadata = memoryBlock.Metadata as VariableBlockMetadata;
-            var driver = variableMetadata?.StorageDriverType;
-
-            // We only capture output written to the workflow itself. Other drivers like blob storage, etc. will be ignored since the foreground context will be loading those.
-            if (driver != typeof(WorkflowStorageDriver) && driver != typeof(WorkflowInstanceStorageDriver) && driver != null)
-                continue;
-
-            var outputValue = activityExecutionContext.Get(memoryBlockReference);
-
-            if (outputValue != null)
-                outputValues[outputDescriptor.Name] = outputValue;
-        }
-
-        return outputValues;
+    {
+        var activityOutputRegister = activityExecutionContext.WorkflowExecutionContext.GetActivityOutputRegister();

```

Comment:
I fear that this "solution" may be too simplistic since the old code clearly went out of it's way to only return values that are backed by `WorkflowInstanceStorageDriver`. I'm not sure that this is a concern now considering that the parent workflow seems to be now resumed in the same thread. Anyway - I'd welcome discussion in this particular change (as well as any other change!).

Reply:
The concern is that when an activity produces a large object, e.g. a file or JSON string of 5 MB, the user might use a different variable storage driver, such as "Blob Storage Driver" that stores the data in Azure Storage.

For this reason, we want to avoid those large values to become part of the output register, given that it will be serialized as well.

Obviously, the current implementation is rather crude and intended to be a temporary hack.
The ultimate solution is to allow for better control of how output is recorded.

For example, for small values, it might be OK to be recorded. The assumption here is that if the storage driver is "Workflow Instance", which stores the value as part of the workflow instance, it's safe to also include it as part of the output register.

For large values, the assumption is that the user will have used the Memory storage driver or a custom one such as "Blob Storage Driver". In this case, we definitely do not want the value to be recorded in the output register.

Now, we actually do want to record _something_ in the output register. In the case of large blobs, a download link could be appropriate.

So perhaps a cleaner approach would be to leverage the associated storage driver to have it produce a value suitable for storing in the output register. This would clean up this code because now we no longer need to test against the storage driver type.

Pending that solution, perhaps for now it might be best to keep the code to only include variables into the output  if those variables are associated with a workflow instance driver.